### ISSUE TITLE
Cursor remains a block even after exiting nvim

### ISSUE THREAD

**Issue by Lightosk** [neovim/neovim#12648]
- `nvim --version`:
```
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.3/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Übersetzt von builduser

Features: +acl +iconv +tui
See ":help feature-compile"

          System-vimrc-Datei: "$VIM/sysinit.vim"
     Voreinstellung für $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: `8.2 (2019 Dec 12 kompiliert am Jun 21 2020 16:11:23)`) behaves differently? Yes.
- Operating system/version: Arch Linux x86_64, updated
- Terminal name/version: Alacritty, zsh
- `$TERM`: alacritty

### Steps to reproduce using `nvim -u NORC`

```
# Be sure to set `cursor.style: underline` in the alacritty.yml.
nvim -u NORC
:q
```

### Actual behaviour
The cursor remains a block after exiting `nvim`, even if it was before starting `nvim` an underline. I have to restart alacritty to return the cursor to an underline.

### Expected behaviour
The cursor turns to an underline again and I am not confused why the terminal feels so different. Vim stays all the time in my normal cursor underline mode.

This has probably a low priority, and I think you forgot to unset the cursor visibility.

**Comment by Lightosk** [comment]
Oh, this is a replicate. I'm sorry.
